main-client.js

```diff
@@ -43,38 +44,42 @@ export const createMeteorNetworkInterface = (givenConfig) => {
   const networkInterface = interfaceToUse(interfaceOptions);
 
   if (config.useMeteorAccounts) {
-    networkInterface.use([{
-      applyMiddleware(request, next) {
-
-        // cookie login token created by meteorhacks:fast-render and caught during server-side rendering by rr:react-router-ssr
-        const { loginToken: cookieLoginToken } = config;
-        // Meteor accounts-base login token stored in local storage, only exists client-side
-        const localStorageLoginToken = Meteor.isClient && Accounts._storedLoginToken();
-
-        // on initial load, prefer to use the token grabbed server-side if existing
-        let currentUserToken = cookieLoginToken || localStorageLoginToken;
-
-        // ...a login token has been passed to the config, however the "true" one is different ⚠️
-        // https://github.com/apollostack/meteor-integration/pull/57/files#r96745502
-        if (Meteor.isClient && cookieLoginToken && cookieLoginToken !== localStorageLoginToken) {
-          // be sure to pass the right token to the request!
-          currentUserToken = localStorageLoginToken;
-        }
-
-        if (!currentUserToken) {
+    // possible cookie login token created by meteorhacks:fast-render 
+    // and passed to the Apollo Client during server-side rendering
+    const { loginToken } = config;
+
+    if (Meteor.isClient && loginToken) {
+      // note: as it's not possible to stop a request,
+      // should this be handled somehow server-side?
+      console.error('[Meteor Apollo Integration] The current user is not handled with your GraphQL requests: you are trying to pass a login token to an Apollo Client instance defined client-side. This is only allowed during server-side rendering, please check your implementation.');
+    } else {
+      networkInterface.use([{
+        applyMiddleware(request, next) {
+          
+          // Meteor accounts-base login token stored in local storage,
+          // only exists client-side as of Meteor 1.4, will exist with Meteor 1.5
+          const localStorageLoginToken = Meteor.isClient && Accounts._storedLoginToken();
+          
+          // define a current user token if existing 
+          // ex: passed during server-side rendering or grabbed from local storage
+          let currentUserToken = localStorageLoginToken || loginToken;
+          
+          if (!currentUserToken) {
+            next();
+            return;
+          }
+          
+          if (!request.options.headers) {
+            // Create the header object if needed.
+            request.options.headers = new Headers();
+          }
+
+          request.options.headers.Authorization = currentUserToken;
```

Comment:
I would say let's use a non-standard header because I think what we're doing is not exactly kosher. We're not authenticating anyone, we're just trying to pass along the auth information from another mechanism into the http mechanism and therefore we are kind of "abusing" that header.

A more standard way would probably be a cookie (but I'm not verry happy and comfortable with using cookies)